pkg/storage/tsdb/config.go

```diff
@@ -292,6 +293,12 @@ func (cfg *TSDBConfig) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.EarlyHeadCompactionMinEstimatedSeriesReductionPercentage, "blocks-storage.tsdb.early-head-compaction-min-estimated-series-reduction-percentage", 10, "When the early compaction is enabled, the early compaction is triggered only if the estimated series reduction is at least the configured percentage (0-100).")
 
 	cfg.HeadCompactionIntervalJitterEnabled = true
+
+	if cfg.WALCompressionEnabled {
+		cfg.WALCompressionType = wlog.CompressionSnappy
+	} else {
+		cfg.WALCompressionType = wlog.CompressionNone
+	}
```

Comment:
This is not correct because it will be evaluated only when the CLI flag is registered and not when it's parsed. Also will not work in case of YAML config.

_I open a PR to fix it._

Reply:
Thanks for the info, will check the PR to see how this should be handled next time!

Reply:
See: https://github.com/grafana/mimir/pull/5597